### DISCUSSION TITLE
Gaming console supports

### DISCUSSION THREAD

**Discussion by dngs365** [mono/mono#21580] [Q&A] [answered]
I'm currently pre-researching for embedding the mono for my C++ application.
[The document](https://www.mono-project.com/docs/about-mono/supported-platforms/) mentions about PS4 and Nintendo Wii, but there is nothing about PS5 and Nintendo Switch.

I also searched about MonoGame and it supports PS5 and Nintendo Switch, they are closed source though.
As I know, the MonoGame is based on the mono, if so, it is natural the mono supports them as well.
However, there is no any clue about it in the document.

Does anyone know about this?

**Comment by lateralusX** [comment, marked as answer]
> I'm currently pre-researching for embedding the mono for my C++ application. [The document](https://www.mono-project.com/docs/about-mono/supported-platforms/) mentions about PS4 and Nintendo Wii, but there is nothing about PS5 and Nintendo Switch.
> 
> I also searched about MonoGame and it supports PS5 and Nintendo Switch, they are closed source though. As I know, the MonoGame is based on the mono, if so, it is natural the mono supports them as well. However, there is no any clue about it in the document.
> 
> Does anyone know about this?

Hi,

Playstation/Xbox/Switch are all NDA based platforms so you will need to request access through described channels in order for us to verify that you are a developer on specific console platforms. mono/mono currently have ports for PS4 and Xbox One and Xbox Series X|S, see https://www.mono-project.com/docs/about-mono/supported-platforms/xbox-series-x-s/ and https://www.mono-project.com/docs/about-mono/supported-platforms/playstation4/ on how to request access. MonoGame can use/uses these ports, but also have a closed source alternative solution maintained by core MonoGame team called Brute. Brute is a custom limited .net runtime not build upon Mono. Since MonoGame console ports are under NDA they follow a similar signup process, https://discord.com/channels/647268145467359274/809369535069159486.

If you currently intend to target Switch, then Brute is the alternative, since there are no official ports of mono/mono targeting that platform at the moment.

First step is to get into needed development program for the platforms you intend to target (needed regardless the solution you will end up using) and then apply for access to existing mono/mono ports as well as MonoGame console ports using the signup processes described above, you will then get access to the repros as well as discord channels where we can discuss current and future plans.

**Reply by dngs365** [reply]
The Brute was covered in veil.
Thanks for details.
